### PR TITLE
go: catch command errors

### LINKED ISSUE #803: Go: break up GetCommandOutput
See discussion here: https://github.com/Originate/git-town/pull/792#discussion_r104321268.

### PR DESCRIPTION
**Pull request by charlierudolph**
resolves #803

### CODE REVIEW

File: lib/config/config.go [hunk 1]
```diff
@@ -52,24 +51,25 @@ func IsFeatureBranch(branchName string) bool {
 
 func IsPerennialBranch(branchName string) bool {
   perennialBranches := GetPerennialBranches()
-  return sort.SearchStrings(perennialBranches, branchName) < len(perennialBranches)
```
Comment (charlierudolph) [inline lib/config/config.go]: [SearchStrings](https://golang.org/pkg/sort/#SearchStrings) tells you where to insert the value into the sorted array and doesn't tell us if the string appears in the list. Haven't run into in any feature tests yet as we I don't know if we have any with both perennial branches and feature branches

Comment (kevgo) [inline lib/config/config.go]: Ah, that's too bad. The Go library is surprisingly minimalistic.

File: lib/config/config.go [hunk 2]
```diff
@@ -52,24 +51,25 @@ func IsFeatureBranch(branchName string) bool {
...
+  return util.DoesStringArrayContain(perennialBranches, branchName)
 }
 
 
 func HasRemoteOrigin() bool {
-  return GetRemoteOriginUrl() != ""
+  return hasRemote("origin")
```
Comment (kevgo) [inline lib/config/config.go]: Do we need this helper at all anymore? Calling 'hasRemote("origin")' is readable enough here in my opinion.

File: lib/config/config.go [hunk 3]
```diff
@@ -52,24 +51,25 @@ func IsFeatureBranch(branchName string) bool {
...
 }
 
 
 func HasRemoteUpstream() bool {
-  return GetRemoteUpstreamUrl() != ""
+  return hasRemote("upstream")
```
Comment (kevgo) [inline lib/config/config.go]: same here

File: lib/config/config.go [hunk 4]
```diff
@@ -52,24 +51,25 @@ func IsFeatureBranch(branchName string) bool {
...
 }
 
 
 func SetParentBranch(branchName, parentBranchName string) {
   storeConfigurationValue(branchName + ".parent", parentBranchName)
 }
 
+
```
Comment (kevgo) [inline lib/config/config.go]: We should start using gofmt. It can convert all files with a single call.

File: lib/config/config.go [hunk 5]
```diff
@@ -79,9 +79,15 @@ func ShouldHackPush() bool {
 
 
 func getConfigurationValue(key string) string {
-  return util.GetCommandOutput([]string{"git", "config", "git-town." + key})
+  namespacedKey := "git-town." + key
+  value := ""
```
Comment (kevgo) [inline lib/config/config.go]: Give the function result a name and get rid of this line?

**Pull request merged**